pkg/asset/cluster/aws/aws.go

```diff
@@ -19,3 +30,73 @@ func Metadata(clusterID, infraID string, config *types.InstallConfig) *aws.Metad
 		}},
 	}
 }
+
+// PreTerraform performs any infrastructure initialization which must
+// happen before Terraform creates the remaining infrastructure.
+func PreTerraform(ctx context.Context, clusterID string) error {
+	privateSubnetString := os.Getenv("TF_VAR_aws_private_subnets") // FIXME: temporary workaround until we get VPCs from the install-config
+	if privateSubnetString == "" {
+		return nil
+	}
+
+	var privateSubnets []string
+	err := json.Unmarshal([]byte(privateSubnetString), &privateSubnets)
+	if err != nil {
+		return errors.Wrapf(err, "parsing private subnets from %q", privateSubnetString)
+	}
+
+	publicSubnetString := os.Getenv("TF_VAR_aws_public_subnets")
+	var publicSubnets []string
+	err = json.Unmarshal([]byte(publicSubnetString), &publicSubnets)
+	if err != nil {
+		return errors.Wrapf(err, "parsing public subnets from %q", publicSubnetString)
+	}
+
+	session, err := awsconfig.GetSession()
+	if err != nil {
+		return err
+	}
+
+	stsClient := sts.New(session)
+	identity, err := stsClient.GetCallerIdentityWithContext(ctx, nil)
+	if err != nil {
+		return errors.Wrap(err, "getting AWS identity")
+	}
+
+	partition, ok := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), *session.Config.Region)
+	if !ok {
+		return errors.Errorf("no partition found for region %q", *session.Config.Region)
+	}
+
+	ec2ARNbase := fmt.Sprintf("arn:%s:ec2:%s:%s", partition.ID(), *session.Config.Region, *identity.Account)
+	arns := []string{
+		fmt.Sprintf("%s:vpc/%s", ec2ARNbase, os.Getenv("TF_VAR_aws_vpc")),
+	}
+	for _, id := range privateSubnets {
+		arns = append(arns, fmt.Sprintf("%s:subnet/%s", ec2ARNbase, id))
+	}
+	for _, id := range publicSubnets {
+		arns = append(arns, fmt.Sprintf("%s:subnet/%s", ec2ARNbase, id))
+	}
```

Comment:
personally, the idenitity from session is too complex _ as it turns out from cred-minter_ that i would like to keep it out of the important code-path for byo-vpc. 

And we only need to tag the subnets, and not VPC so direct calls to tag subnets are more preferable here, unless you can find the arn from API rather than creating them ourselves please.

Reply:
> personally, the idenitity from session is too complex...

I'm punting all of the tagging stuff to follow-up PRs.